### PR TITLE
polish the config region_split_size

### PR DESCRIPTION
**Pull request by tonyxuqqi**
Signed-off-by: qi.xu <tonxuqi@outlook.com>

### What is changed and how it works?

Issue Number: Ref #12842

What's Changed:

```commit-message
materialize the region_split_size in optimize_for
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore/src/coprocessor/config.rs [hunk 1]
```diff
@@ -130,8 +130,12 @@ impl Config {
 
     pub fn optimize_for(&mut self, raftstore_v2: bool) {
         // overwrite the default region_split_size when it's multi-rocksdb
-        if raftstore_v2 && self.region_split_size.is_none() {
-            self.region_split_size = Some(ReadableSize::mb(RAFTSTORE_V2_SPLIT_SIZE_MB));
+        if self.region_split_size.is_none() {
```
Comment (BusyJay) [inline components/raftstore/src/coprocessor/config.rs:133]: Why the change?

**Review by cosven — APPROVED** [review]

File: components/raftstore/src/coprocessor/config.rs (hunk 1 above)
Comment (cosven) [reply components/raftstore/src/coprocessor/config.rs:133]: This field used to be a string, and other tools (like lightning) use it to confirm what region size is. A previous PR changed it to a Option, so the field value maybe a null object, which causes compatibility issue.
image

Maybe we should add a unit test to ensure this field is a string. :thinking:

**Review by BusyJay — APPROVED** [review]

File: components/raftstore/src/coprocessor/config.rs (hunk 1 above)
Comment (tonyxuqqi) [reply components/raftstore/src/coprocessor/config.rs:133]: The test below could make sure it returns the actual value without explicitly setting the field.

**Review by nolouch — APPROVED** [review]
